app/models/semester_period.rb

```diff
@@ -0,0 +1,42 @@
+require 'sirius/enums/semester_period_type'
+require 'parity'
+
+class SemesterPeriod < Sequel::Model
+
+  many_to_one :faculty_semester
+
+  def type
+    Sirius::SemesterPeriodType.from_numeric(super)
+  end
+
+  def type=(new_type)
+    super Sirius::SemesterPeriodType.to_numeric(new_type)
+  end
+
+  def first_week_parity
+    if super
+      Parity.from_numeric(super)
```

Comment:
Well, we use this pattern everywhere. Perhaps you could use an instance var here, but perhaps not, if sequel handles this through method_missing. I think this is exactly the point where it is better to treat ORM as black box and stick with the recommended pattern.